### ISSUE TITLE
Faster joins: check how incoming events are authed during the partial-state phase

### ISSUE THREAD

**Issue by richvdh** [matrix-org/synapse#12644]
I/we need to revisit how incoming events are authed while we lack full state. When I was testing, it looked like we were relying on Synapse's unspecced `_update_auth_events_for_auth` behaviour. If we're going to rely on that, we'll have to understand it and spec it.

A subtask of this is to figure out if we need power events upfront (eg, banned users) to get reliable state.

**Comment by erikjohnston** [comment]
Did we not decide that we should try ripping the `_update_auth_events_for_auth` code out?

**Comment by richvdh** [comment]
> Did we not decide that we should try ripping the `_update_auth_events_for_auth` code out?

We talked about it back in a previous tranche of event-auth work (see https://github.com/matrix-org/synapse/issues/9595#issuecomment-904662133 etc), but I think we decided we needed to convince ourselves it wasn't important, first.  The work for #9595 ended up significantly simplifying it, and documenting its interface a bit, but never got as far as completely ripping it out.

The problem we have now is that the faster-joins work appears to be *relying* on `_update_auth_events_for_auth`, so we now need to re-consider whether we still want to rip it out, or do something else for faster-joins.
